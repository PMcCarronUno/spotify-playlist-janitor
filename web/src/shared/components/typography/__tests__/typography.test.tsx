import { render } from "@testing-library/react";

import { Title, SubTitle, Text, SecondaryText, SubText } from "..";

describe("<Title />", () => {
  let container: HTMLElement;
  let props: any;

  beforeEach(() => {
    props = {
      id: "title-test",
      "data-testid": "title-test-data-testid",
    };
    ({ container } = render(<Title {...props}>Title Text</Title>));
  });

  it("should render Title component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("<SubTitle />", () => {
  let container: HTMLElement;
  let props: any;

  beforeEach(() => {
    props = {
      id: "subtitle-test",
      "data-testid": "subtitle-test-data-testid",
    };
    ({ container } = render(<SubTitle {...props}>Subtitle Text</SubTitle>));
  });

  it("should render SubTitle component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("<Text />", () => {
  let container: HTMLElement;
  let props: any;

  beforeEach(() => {
    props = {
      id: "text-test",
      "data-testid": "text-test-data-testid",
    };
    ({ container } = render(<Text {...props}>Text</Text>));
  });

  it("should render Text component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("<SecondaryText />", () => {
  let container: HTMLElement;
  let props: any;

  beforeEach(() => {
    props = {
      id: "secondary-text-test",
      "data-testid": "secondary-text-test-data-testid",
    };
    ({ container } = render(<SecondaryText {...props}>Secondary Text</SecondaryText>));
  });

  it("should render SecondaryText component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("<SubText />", () => {
  let container: HTMLElement;
  let props: any;

  beforeEach(() => {
    props = {
      id: "sub-text-test",
      "data-testid": "sub-text-test-data-testid",
    };
    ({ container } = render(<SubText {...props}>Sub Text</SubText>));
  });

  it("should render SubText component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});