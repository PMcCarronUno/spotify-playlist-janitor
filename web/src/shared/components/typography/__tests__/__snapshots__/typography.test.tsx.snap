// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryText /> should render SecondaryText component 1`] = `
<div
  class="sc-hLQTFJ gjIzVE"
  data-testid="secondary-text-test-data-testid"
  id="secondary-text-test"
>
  Secondary Text
</div>
`;

exports[`<SubText /> should render SubText component 1`] = `
<div
  class="sc-eDLKEg ebFBpI"
  data-testid="sub-text-test-data-testid"
  id="sub-text-test"
>
  Sub Text
</div>
`;

exports[`<SubTitle /> should render SubTitle component 1`] = `
<div
  class="sc-guDMob jLqGJD"
  data-testid="subtitle-test-data-testid"
  id="subtitle-test"
>
  Subtitle Text
</div>
`;

exports[`<Text /> should render Text component 1`] = `
<div
  class="sc-dmyDmy dcXriK"
  data-testid="text-test-data-testid"
  id="text-test"
>
  Text
</div>
`;

exports[`<Title /> should render Title component 1`] = `
<div
  class="sc-beyTiQ gCCLUC"
  data-testid="title-test-data-testid"
  id="title-test"
>
  Title Text
</div>
`;
